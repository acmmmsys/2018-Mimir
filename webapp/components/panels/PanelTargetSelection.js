import React from 'react';
import { connect } from 'react-redux';
import { compose, withProps, nest, withHandlers, lifecycle, defaultProps } from 'recompose';
import { requestCnnClasses, selectCnnClass, requestImageVisualization } from 'actions';

import Panel from 'layout/Panel'
import TableSelector from 'components/ui/TableSelector';
import {  } from 'actions';

const enhance = compose(
    connect(
        state => ({
            classes: state.cnn.classes,
            imageId: state.cnn.selectedImageId,
            layerId: state.cnn.selectedLayer,
            classId: state.cnn.selectedClass,
            selectedRow: Number(state.cnn.selectedClass),
            classification: state.cnn.classifications[state.cnn.selectedImageId],
            loading: state.loading.classes
        }),
        dispatch => ({
            requestCnnClasses() {
                dispatch(requestCnnClasses());
            },
            selectVisualizationTarget(imageId, layerId, classId){
                dispatch(selectCnnClass(classId));
                dispatch(requestImageVisualization(imageId, layerId, classId));
            },
        })
    ),
    defaultProps({
        classification: [],
    }),
    withHandlers({
        onClick: props => rowIndex => {
            props.selectVisualizationTarget(
                props.imageId,
                props.layerId,
                Object.keys(props.classes)[rowIndex]
            );
        },
    }),
    withProps(props => ({
        isEmpty: !props.imageId,
        header: [ 'Class', 'Probability' ],
        content: props.classes.map(label => [label, `${(Number(props.classification[label]) * 100).toFixed(3).replace(/\.0+$/,'')}%`]),
        emptyMessage: 'List of targets will be listed here',
    })),
    lifecycle({
        componentWillMount()Â {
            this.props.requestCnnClasses();     
        },
    })
);

export default enhance(nest(Panel, TableSelector));