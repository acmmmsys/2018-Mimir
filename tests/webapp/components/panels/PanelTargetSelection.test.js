import React from 'react';
import PanelTargetSelection from '../../../../webapp/components/panels/PanelTargetSelection'
import renderer from 'react-test-renderer';
import configureStore from 'redux-mock-store'

const mockStore = configureStore();

let store;
let wrapper;

test('PanelTargetSelection does render', () => {
  const wrapper = shallow(
      <PanelTargetSelection store={mockStore(initialState)}>This is a test!</PanelTargetSelection>
  );
  expect(wrapper).toMatchSnapshot();
});
