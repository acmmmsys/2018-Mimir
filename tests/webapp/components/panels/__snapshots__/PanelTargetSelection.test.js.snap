// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PanelTargetSelection does render 1`] = `
<defaultProps(withHandlers(withProps(lifecycle(nest(defaultProps(branch(branch(Panel))), TableSelector)))))
  classId={6}
  classes={Array []}
  classification={Array []}
  layerId="block5_conv4"
  loading={false}
  requestCnnClasses={[Function]}
  selectVisualizationTarget={[Function]}
  selectedRow={6}
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
  storeSubscription={
    Subscription {
      "listeners": Object {
        "clear": [Function],
        "get": [Function],
        "notify": [Function],
        "subscribe": [Function],
      },
      "onStateChange": [Function],
      "parentSub": undefined,
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "unsubscribe": [Function],
    }
  }
>
  This is a test!
</defaultProps(withHandlers(withProps(lifecycle(nest(defaultProps(branch(branch(Panel))), TableSelector)))))>
`;
